runtime/compilers/rillv1/parser.go

```diff
@@ -345,6 +393,27 @@ func (p *Parser) parsePaths(ctx context.Context, paths []string) error {
 		return ErrInvalidProject
 	}
 
+	// As a special case, we need to check that there aren't any sources and models with the same name.
+	// NOTE 1: We always attach the error to the model when there's a collision.
+	// NOTE 2: Using a map since the two-way check (necessary for reparses) may match the same resource twice.
+	modelsWithNameErrs := make(map[ResourceName]string)
```

Comment:
I cant find the place where this check is happening for the same `kind/name` with different stem. Or do we not allow different stems to have same name be different resource?

Reply:
Discussed offline, will probably want to make some refactors to avoid this in a separate PR